src/Eagle.ts

```diff
@@ -2104,14 +2110,23 @@ export class Eagle {
     addNodeToLogicalGraph = (node : Node) : void => {
         //console.log("addNodeToLogicalGraph()", node.getName(), node.getCategory(), node.getInputPorts().length, node.getOutputPorts().length, node.getFields().length);
 
+        
         // get new position for node
-        let pos = this.getNewNodePosition();
+        if (Eagle.nodeDropLocation.x == 0 && Eagle.nodeDropLocation.y == 0){
```

Comment:
I think you are using (Eagle.nodeDropLocation.x === 0 && Eagle.nodeDropLocation.y === 0) to mean that the node was not placed using a drag-and-drop, that the button was clicked instead. If I am right, then I think this could be more explicit, perhaps have a boolean variable that records whether a drag-and-drop occurred and hence the Eagle.nodeDropLocation should be used.